test/cql-pytest/test_permissions.py

```diff
@@ -309,19 +322,28 @@ def test_grant_revoke_alter_udf_permissions(cassandra_bug, cql):
                             function=lambda: user_session.execute(f"CREATE OR REPLACE FUNCTION {keyspace}.{fun} {fun_body}"))
 
 # Test that granting permissions on non-existent UDFs fails
-def test_grant_perms_on_nonexistent_udf(cql):
+# This test is marked cassandra_bug because it exposes a small error-path
+# bug in their parser (see comment below).
+def test_grant_perms_on_nonexistent_udf(cql, cassandra_bug):
     keyspace = "ks"
     fun_name = "fun42"
     with new_user(cql) as username:
         grant(cql, 'EXECUTE', 'ALL FUNCTIONS', username)
         revoke(cql, 'EXECUTE', 'ALL FUNCTIONS', username)
         with pytest.raises(InvalidRequest):
             grant(cql, 'EXECUTE', f'ALL FUNCTIONS IN KEYSPACE {keyspace}', username)
-        with pytest.raises(InvalidRequest):
+        # When the keyspace is non-existant, Scylla returns InvalidRequest
+        # and Cassandra returns ConfigurationException. Let's allow both.
+        with pytest.raises((InvalidRequest, ConfigurationException)):
```

Comment:
Allowing both opens door for scylla to also behave "incorrectly", i.e. the same way cassandra does. Perhaps making this check dependent on the db, i.e. scylla or cassandra, would be safer. So if it's scylla -> expect InvalidRequest, if cassandra -> congi exception.

Reply:
I agree about the first sentence, but to be honest, I don't know if either one of them should be considered incorrect. It's not documented anywhere I'm aware of, and I don't think we will really care if in some future version we or Cassandra change our mind on this issue.

Reply:
Fair enough, let's leave it as is